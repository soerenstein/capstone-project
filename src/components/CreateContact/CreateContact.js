import React, { useState } from 'react'
import styled from 'styled-components'

export default function CreateContact() {
  const [contacts, setContacts] = useState([])
  const [disabledButton, setDisabledButton] = useState(true)

  return (
    <div>
      <FormStyled onSubmit={handleSubmit}>
        <LabelWrapper>
          <LabelStyled>
            Vorname
            <input
              autoFocus
              name="firstName"
              id="firstName"
              type="text"
              maxLength="40"
              value={contacts.firstName}
              onChange={handleChange}
              placeholder="Max"
            />
          </LabelStyled>
          <LabelStyled>
            Nachname
            <input
              name="lastName"
              id="lastName"
              type="text"
              maxLength="40"
              value={contacts.lastName}
              onChange={handleChange}
              placeholder="Mustermann"
            />
          </LabelStyled>
        </LabelWrapper>
        <LabelStyled>
          Unternehmen
          <input
            name="company"
            id="company"
            type="text"
            onChange={handleChange}
            placeholder="Universal GmbH"
          />
        </LabelStyled>
        <LabelStyled>
          Telefon
          <input name="phone" type="tel" id="phone" />
        </LabelStyled>
        <LabelStyled>
          E-Mail
          <input name="mail" id="mail" />
        </LabelStyled>

        <LabelStyled>
          Adresse
          <input name="zip" type="text" id="street" placeholder="Straße" />
          <LabelWrapper>
            <input name="zip" type="number" id="zip" placeholder="PLZ" />
            <input name="city" type="text" id="city" placeholder="Stadt" />
          </LabelWrapper>
        </LabelStyled>
        <LabelStyled>
          Notiz
          <input name="note" type="text" id="note" />
        </LabelStyled>
        <LabelStyled>
          Kategorie
          <input
            name="category"
            type="text"
            id="category"
            placeholder="Kunde, Dienstleister"
          />
        </LabelStyled>
        <SubmitButton disabled={disabledButton}>
          Kontakt hinzufügen
        </SubmitButton>
      </FormStyled>
    </div>
  )

  function handleChange() {
    setDisabledButton(contacts.firstName === '' ? true : false)
  }

  function handleSubmit(event) {
    event.preventDefault()
    const form = event.target
    const contact = [
      {
        firstName: form.firstName.value,
        lastName: form.lastName.value,
        company: form.company.value,
        phone: form.phone.value,
        mail: form.mail.value,
        street: form.street.value,
        zip: form.zip.value,
        city: form.city.value,
        note: form.note.value,
        category: form.category.value,
      },
    ]
    console.log(contact)
    setContacts([contact, ...contacts])
    setDisabledButton(true)
    form.reset()
    form[0] && form[0].focus()
  }
}

const FormStyled = styled.form`
  display: grid;
  gap: 10px;
  padding: 20px;
`
const LabelWrapper = styled.div`
  display: grid;
  grid-template-columns: auto auto;
  gap: 10px;
`

const LabelStyled = styled.label`
  display: grid;
  gap: 10px;
`
const SubmitButton = styled.button``
