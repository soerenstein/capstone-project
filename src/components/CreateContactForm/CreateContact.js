import React from 'react'
import CreateContactForm from './CreateContactForm'
import Header from '../Header/Header'

export default function CreateContact({ onSubmit }) {
    return (
        <div>
            <Header />
            <CreateContactForm />
        </div>
    )
}
