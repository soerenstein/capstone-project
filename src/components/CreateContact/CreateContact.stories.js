import React from 'react'
import CreateContact from './CreateContact'

export default {
    title: 'CreateContact',
    component: CreateContact,
}

export const Form = () => <CreateContact />