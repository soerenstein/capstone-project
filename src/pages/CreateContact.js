import React from 'react'
import CreateContactForm from '../components/CreateContactForm/CreateContactForm'

export default function CreateContact({ onSubmit }) {
  return <CreateContactForm onSubmit={onSubmit} />
}
