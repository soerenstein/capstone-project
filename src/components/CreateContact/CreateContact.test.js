import React from 'react'
import { render } from '@testing-library/react'
import CreateContact from './CreateContact'

describe('The form', () => {
    it('renders correctly', () => {
        const form = render(<CreateContact />)
        expect(form).toMatchSnapshot()
    })
})