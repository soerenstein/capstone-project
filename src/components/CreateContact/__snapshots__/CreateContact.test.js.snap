// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The form renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <form
          class="sc-AxjAm htDRZe"
        >
          <div
            class="sc-AxirZ erNbfn"
          >
            <label
              class="sc-AxiKw bMvXWd"
            >
              Vorname
              <input
                id="firstName"
                maxlength="40"
                name="firstName"
                placeholder="Max"
                type="text"
                value=""
              />
            </label>
            <label
              class="sc-AxiKw bMvXWd"
            >
              Nachname
              <input
                id="lastName"
                maxlength="40"
                name="lastName"
                placeholder="Mustermann"
                type="text"
                value=""
              />
            </label>
          </div>
          <label
            class="sc-AxiKw bMvXWd"
          >
            Unternehmen
            <input
              id="company"
              name="company"
              placeholder="Universal GmbH"
              type="text"
            />
          </label>
          <label
            class="sc-AxiKw bMvXWd"
          >
            Telefon
            <input
              id="phone"
              name="phone"
              type="tel"
            />
          </label>
          <label
            class="sc-AxiKw bMvXWd"
          >
            E-Mail
            <input
              id="mail"
              name="mail"
            />
          </label>
          <label
            class="sc-AxiKw bMvXWd"
          >
            Adresse
            <input
              id="street"
              name="zip"
              placeholder="Straße"
              type="text"
            />
            <div
              class="sc-AxirZ erNbfn"
            >
              <input
                id="zip"
                name="zip"
                placeholder="PLZ"
                type="number"
              />
              <input
                id="city"
                name="city"
                placeholder="Stadt"
                type="text"
              />
            </div>
          </label>
          <label
            class="sc-AxiKw bMvXWd"
          >
            Notiz
            <input
              id="note"
              name="note"
              type="text"
            />
          </label>
          <label
            class="sc-AxiKw bMvXWd"
          >
            Kategorie
            <input
              id="category"
              name="category"
              placeholder="Kunde, Dienstleister"
              type="text"
            />
          </label>
          <button
            class="sc-AxhCb ccMvAG"
            disabled=""
          >
            Kontakt hinzufügen
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <form
        class="sc-AxjAm htDRZe"
      >
        <div
          class="sc-AxirZ erNbfn"
        >
          <label
            class="sc-AxiKw bMvXWd"
          >
            Vorname
            <input
              id="firstName"
              maxlength="40"
              name="firstName"
              placeholder="Max"
              type="text"
              value=""
            />
          </label>
          <label
            class="sc-AxiKw bMvXWd"
          >
            Nachname
            <input
              id="lastName"
              maxlength="40"
              name="lastName"
              placeholder="Mustermann"
              type="text"
              value=""
            />
          </label>
        </div>
        <label
          class="sc-AxiKw bMvXWd"
        >
          Unternehmen
          <input
            id="company"
            name="company"
            placeholder="Universal GmbH"
            type="text"
          />
        </label>
        <label
          class="sc-AxiKw bMvXWd"
        >
          Telefon
          <input
            id="phone"
            name="phone"
            type="tel"
          />
        </label>
        <label
          class="sc-AxiKw bMvXWd"
        >
          E-Mail
          <input
            id="mail"
            name="mail"
          />
        </label>
        <label
          class="sc-AxiKw bMvXWd"
        >
          Adresse
          <input
            id="street"
            name="zip"
            placeholder="Straße"
            type="text"
          />
          <div
            class="sc-AxirZ erNbfn"
          >
            <input
              id="zip"
              name="zip"
              placeholder="PLZ"
              type="number"
            />
            <input
              id="city"
              name="city"
              placeholder="Stadt"
              type="text"
            />
          </div>
        </label>
        <label
          class="sc-AxiKw bMvXWd"
        >
          Notiz
          <input
            id="note"
            name="note"
            type="text"
          />
        </label>
        <label
          class="sc-AxiKw bMvXWd"
        >
          Kategorie
          <input
            id="category"
            name="category"
            placeholder="Kunde, Dienstleister"
            type="text"
          />
        </label>
        <button
          class="sc-AxhCb ccMvAG"
          disabled=""
        >
          Kontakt hinzufügen
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
